File: back-end/src/Routes/userRoutes.js
import express from 'express'
import userController from "../Controllers/userController.js"
import authController from "../Controllers/authController.js"

const router = express.Router()

router
    .get("/getAllUsers", userController.getAll)
    .post("/createUser", userController.createUser)
    .post("/login", authController.login)
    .get("getUserById/:id", userController.getUserById)
    .patch("/updateUserById/:id", userController.updateUserById)
    .delete("/deleteUserById/:id", userController.deleteUserById)

export default router

// authController.checkToken, 

File: back-end/src/Routes/userRoutes.py
from flask import Blueprint
from Controllers.userController import index, create, read, update, destroy,login

user_bp = Blueprint('user_bp', __name__)

user_bp.route('/', methods=['GET'])(index)
user_bp.route('/', methods=['POST'])(create)
user_bp.route('/<int:id>', methods=['GET'])(read)
user_bp.route('/<int:id>', methods=['PUT', 'PATCH'])(update)
user_bp.route('/<int:id>', methods=['DELETE'])(destroy)
user_bp.route('/login', methods=['POST'])(login)
